apis/r/R/SOMAExperimentAxisQuery.R

```diff
@@ -169,6 +169,161 @@ SOMAExperimentAxisQuery <- R6::R6Class(
+    #' labels depends on the selected `collection`:
+    #'
+    #' | Collection | `obs_index`          | `var_index`          |
+    #' |-----------:|:---------------------|:---------------------|
+    #' | `X`        | row names            | column names         |
+    #' | `obsm`     | row names            | ignored              |
+    #' | `varm`     | ignored              | row names            |
+    #' | `obsp`     | row and column names | ignored              |
+    #' | `varp`     | ignored              | row and column names |
+    #' @return A [`Matrix::sparseMatrix-class`]
+    to_sparse_matrix = function(
+      collection, layer_name, obs_index = NULL, var_index = NULL
+    ) {
+      stopifnot(
+        assert_subset(
+          x = collection,
+          y = c("X", "obsm", "varm", "obsp", "varp"),
+          type = "collection"
+        ),
+
+        "Must specify a single layer name" = is_scalar_character(layer_name),
+        assert_subset(layer_name, self$ms[[collection]]$names(), "layer"),
+
+        "Must specify a single obs index" =
+          is.null(obs_index) || is_scalar_character(obs_index),
+        assert_subset(obs_index, self$obs_df$colnames(), "column"),
+
+        "Must specify a single var index" =
+          is.null(var_index) || is_scalar_character(var_index),
+        assert_subset(var_index, self$var_df$colnames(), "column")
+      )
+
+      # Retrieve and validate obs/var indices
+      obs_labels <- var_labels <- NULL
+      if (!is.null(obs_index)) {
+        if (collection %in% c("varm", "varp")) {
+          spdl::warn("The obs_index is ignored for {} collections", collection)
+        } else {
+          obs_labels <- self$obs(column_names = obs_index)[[1]]$as_vector()
+        }
+        stopifnot(
+          "All obs_index values must be unique" = anyDuplicated(obs_labels) == 0
+        )
+      }
+
+      if (!is.null(var_index)) {
+        if (collection %in% c("obsm", "obsp")) {
+          spdl::warn("The var_index is ignored for {} collections", collection)
+        } else {
+          var_labels <- self$var(column_names = var_index)[[1]]$as_vector()
+        }
+        stopifnot(
+          "All var_index values must be unique" = anyDuplicated(var_labels) == 0
+        )
+      }
+
+      # Construct coordinates
+      coords <- switch(collection,
+        X = list(
+          soma_dim_0 = self$obs_joinids(),
+          soma_dim_1 = self$var_joinids()
+        ),
+        obsm = list(
+          soma_dim_0 = self$obs_joinids()
+        ),
+        varm = list(
+          soma_dim_0 = self$var_joinids()
+        ),
+        obsp = list(
+          soma_dim_0 = self$obs_joinids(),
+          soma_dim_1 = self$obs_joinids()
+        ),
+        varp = list(
+          soma_dim_0 = self$var_joinids(),
+          soma_dim_1 = self$var_joinids()
+        )
+      )
+
+      # TODO: Coords must be vectors until read() supports arrow arrays
```

Comment:
Small comment: this one is a constant pain point throughout many use cases, we should make it a high priority for post 1.0. In this case support for arrow arrays would be great, but ultimately a lightweight representation of ranges would be ideal

Reply:
@pablo-gar please file a GitHub issue to track -- thanks!

Reply:
https://github.com/single-cell-data/TileDB-SOMA/issues/1479